compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/context/goto_ctx.rs

```diff
@@ -70,10 +74,16 @@ impl<'tcx> GotocCtx<'tcx> {
 
 /// Getters
 impl<'tcx> GotocCtx<'tcx> {
-    pub fn crate_name(&self) -> String {
+    /// The short crate name without versioning information.
+    pub fn short_crate_name(&self) -> String {
         self.tcx.crate_name(LOCAL_CRATE).to_string()
     }
 
+    /// The full crate name including versioning info
+    pub fn full_crate_name(&self) -> &str {
+        &self.full_crate_name
```

Comment:
Why do we cache this one and not the other one?

Reply:
One we pull directly from MIR, the other will build from two parts. Maybe with the `to_string` in the first case it's the same, though?